bundles/org.openhab.binding.digitalstrom/src/main/java/org/openhab/binding/digitalstrom/internal/lib/manager/impl/DeviceStatusManagerImpl.java

```diff
@@ -387,18 +380,16 @@ public void run() {
                             DeviceStatusListener.DEVICE_DISCOVERY, device.getDSID().getValue());
                 } else {
                     logger.debug(
-                            "The device-Discovery is not registrated, can't inform device discovery about removed device.");
+                            "The device-Discovery is not registered, can't inform device discovery about removed device.");
                 }
             }
 
             if (!trashDevices.isEmpty() && (lastBinCheck + config.getBinCheckTime() < System.currentTimeMillis())) {
-                for (TrashDevice trashDevice : trashDevices) {
-                    if (trashDevice.isTimeToDelete(Calendar.getInstance().get(Calendar.DAY_OF_YEAR))) {
-                        logger.debug("Found trashDevice that have to delete!");
-                        trashDevices.remove(trashDevice);
-                        logger.debug("Delete trashDevice: {}", trashDevice.getDevice().getDSID().getValue());
-                    }
-                }
+                trashDevices.stream().filter(d -> d.isTimeToDelete(Calendar.getInstance().get(Calendar.DAY_OF_YEAR)))
+                        .forEach(trashDevice -> {
+                            trashDevices.remove(trashDevice);
+                            logger.debug("Deleted trashDevice: {}", trashDevice.getDevice().getDSID().getValue());
+                        });
```

Comment:
Removing values during iteration is asking for trouble. Although CopyOnWriteArrayList would safely handle such a case I'd prefer if `removeIf` was used instead since it is not implementation dependent.
```suggestion
                trashDevices.removeIf(trashDevice -> {
                    if (trashDevice.isTimeToDelete(Calendar.getInstance().get(Calendar.DAY_OF_YEAR)))){
                        logger.debug("Deleted trashDevice: {}", trashDevice.getDevice().getDSID().getValue());
                        return true;
                    }
                    return false;
                });
```